presto-prometheus/src/main/java/com/facebook/presto/plugin/prometheus/PrometheusQueryResponse.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.prometheus;
+
+import com.facebook.presto.spi.PrestoException;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.plugin.prometheus.PrometheusErrorCode.PROMETHEUS_PARSE_ERROR;
+import static com.fasterxml.jackson.core.JsonToken.FIELD_NAME;
+import static java.util.Collections.singletonList;
+
+public class PrometheusQueryResponse
+{
+    private boolean status;
+
+    private String error;
+    private String errorType;
+    private String resultType;
+    private String result;
+    private List<PrometheusMetricResult> results;
+
+    public PrometheusQueryResponse(InputStream response)
+            throws IOException
+    {
+        parsePrometheusQueryResponse(response);
+    }
+
+    private void parsePrometheusQueryResponse(InputStream response)
+            throws IOException
+    {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new JavaTimeModule());
+        JsonParser parser = new JsonFactory().createParser(response);
+        while (!parser.isClosed()) {
+            JsonToken jsonToken = parser.nextToken();
+            if (FIELD_NAME.equals(jsonToken)) {
+                if (parser.getCurrentName().equals("status")) {
```

Comment:
nit: I would extract "status", "success" and "resultType" into constants, looks like they occurred a couple of times in this PR.  Your call.